packages/babel-parser/src/tokenizer/state.js

```diff
@@ -59,6 +59,7 @@ export default class State {
   // Flags to track
   inParameters: boolean = false;
   maybeInArrowParameters: boolean = false;
+  maybeInAsyncArrowHead: boolean = false;
```

Comment:
It's not clear to me what the difference between these two flags should be. Isn't the arrow head its parameters?

Reply:
The `maybeInAsyncArrowHead` lives across function declarations.
```js
async (foo = function (await) {}) => {}
```
When parsing `await`, `maybeInAsyncArrowHead` is `true` but `maybeInArrowParameters` is `false`.

Reply:
We should add a comment in the code explaining it 👍

Reply:
@nicolo-ribaudo Done. Let me know if we can make it better.